sw/device/lib/dif/dif_plic.h

```diff
@@ -81,7 +81,23 @@ typedef enum dif_plic_irq_id {
   kDifPlicIrqIdNmiGenEsc1 = 61,         /**< NMI Gen escalation interrupt 1. */
   kDifPlicIrqIdNmiGenEsc2 = 62,         /**< NMI Gen escalation interrupt 2. */
   kDifPlicIrqIdNmiGenEsc3 = 63,         /**< NMI Gen escalation interrupt 3. */
-  kDifPlicIrqIdLast = kDifPlicIrqIdNmiGenEsc3, /**< The last valid IRQ ID. */
+  kDifPlicIrqIdUsbDevPktReceived = 64,  /**< USB packet received (OUT/SETUP). */
+  kDifPlicIrqIdUsbDevPktSent = 65,      /**< USB packet sent (IN). */
+  kDifPlicIrqIdUsbDevDisconnected = 66, /**< USB link disconn (VBUS lost). */
+  kDifPlicIrqUsbDevHostLost = 67,       /*<< USB frame not received in time. */
+  kDifPlicIrqUsbDevLinkReset = 68,      /*<< USB link reset. */
+  kDifPlicIrqUsbDevLinkSuspend = 69,    /*<< USB link suspend. */
+  kDifPlicIrqUsbDevLinkResume = 70,     /*<< USB link resume. */
+  kDifPlicIrqUsbDevAvEmpty = 71,    /*<< USB AvBuffer FIFO empty (OUT/SETUP). */
+  kDifPlicIrqUsbDevRxFull = 72,     /*<< USB Receive FIFO full (OUT/SETUP). */
+  kDifPlicIrqUsbDevAvOverflow = 73, /*<< USB AvBuffer FIFO overflow. */
+  kDifPlicIrqUsbDevLinkInErr = 74,  /*<< USB IN error. */
+  kDifPlicIrqUsbDevRxCrcErr = 75,   /*<< USB CRC error. */
+  kDifPlicIrqUsbDevRxPidErr = 76,   /*<< USB PID invalid error. */
+  kDifPlicIrqUsbDevRxBitstuffErr = 77, /*<< USB bitstuffing invalid error. */
+  kDifPlicIrqUsbDevFrame = 78,     /*<< USB frame number updated, SOF valid. */
+  kDifPlicIrqUsbDevConnected = 79, /*<< USB connected (VBUS applied). */
+  kDifPlicIrqIdLast = kDifPlicIrqUsbDevConnected, /**< The last valid IRQ ID. */
 } dif_plic_irq_id_t;
```

Comment:
I'm not sure if autogen'ing the enum is the best idea, but I think it might be nice to have autogen'd values (like register offset headers) that we can use for the enum variant values.

Reply:
@mcy I completely agree, I will open an issue. Meanwhile we could merge this, to make PLIC up-to-date.